samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemAdmin.scala

```diff
@@ -502,24 +500,32 @@ class KafkaSystemAdmin(
   }
 
   /**
-   * Exception to be thrown when the change log stream creation or validation has failed
-   */
+    * Exception to be thrown when the change log stream creation or validation has failed
+    */
   class KafkaChangelogException(s: String, t: Throwable) extends SamzaException(s, t) {
     def this(s: String) = this(s, null)
   }
 
   override def createChangelogStream(topicName: String, numKafkaChangelogPartitions: Int) = {
-    createTopicInKafka(topicName, numKafkaChangelogPartitions)
-    validateChangelogStream(topicName, numKafkaChangelogPartitions)
+    val topicMeta = topicMetaInformation.getOrElse(topicName, throw new KafkaChangelogException("Unable to find topic information for topic " + topicName))
+    val spec = new KafkaStreamSpec(topicName, topicName, systemName, numKafkaChangelogPartitions, topicMeta.replicationFactor, topicMeta.kafkaProps)
+
+    if (createStream(spec)) {
+      info("Created stream %s." format topicName)
+    } else {
+      info("Changelog stream %s already exists." format topicName)
```

Comment:
The javadoc for createStream() needs to include a bit more info regarding to the failure cases. Under which condition should the user expect a "false"? Is there any exception thrown instead of true/false result?

The log line here seems to indicate that "false" means the topic already exists. Is it always the case?